public/app/features/alerting/unified/components/import-to-gma/ConfirmConvertModal.tsx

```diff
@@ -58,9 +59,37 @@ export const ConfirmConversionModal = ({ isOpen, onDismiss }: ModalProps) => {
     'ruleGroup',
     'targetDatasourceUID',
   ]);
-  const { rulerRules } = useGetRulerRules(selectedDatasourceName || undefined);
+  // we need to skip fetching if the modal is not open
+  const dataSourceToFetch = isOpen ? selectedDatasourceName : undefined;
+  const targetFolderToFetch = isOpen ? targetFolder : undefined;
+  const { rulerRules, isLoading: isloadingCloudRules } = useGetRulerRules(dataSourceToFetch || undefined);
+  const { currentData: rulerNamespace, isLoading: loadingTargetGroups } = useFetchGroupsForFolder(
+    targetFolderToFetch?.uid ?? ''
+  );
+
+  const targetFolderHasRules = Boolean(rulerNamespace && hasRules(rulerNamespace));
+
   const [convert] = convertToGMAApi.useConvertToGMAMutation();
   const notifyApp = useAppNotification();
+
+  if (isloadingCloudRules || loadingTargetGroups) {
+    return (
+      <Modal
+        isOpen={isOpen}
+        title={t('alerting.import-to-gma.confirm-modal.loading', 'Loading...')}
+        onDismiss={onDismiss}
+        onClickBackdrop={onDismiss}
+      >
+        <Text>
+          {t(
+            'alerting.import-to-gma.confirm-modal.loading-body',
+            'Preparing data to be imported.This can take a while...'
```

Comment:
depending on the target folder content or the ruler rules to be imported, this can take some seconds...or more...?